Removed JQuery dependency

cyruscook opened the pull request:
I wanted to use this on a website I'm building that doesn't have JQuery, so I rewrote the plugin to use vanilla JS.


It's seems to be working fine, and I'm implementing it into my own project right now, however I wondered if you'd like to include my changes in the base repo? There's practically no reason to use JQuery as it takes 100ms to load (on my computer) and nearly everything is achievable without it.


If you do want to use my code, please tell me as I will have to make some changes. 
- First off, I use tabs instead of spaces, but I can revert that. 
- Secondly, I put my `{` on a new line, however I can revert that too. 
- Thirdly, I couldn't get the animations to be the same when animating it with vanilla JS. I'm not too sure why this is, however there will be a difference both in speed and interpolation when using the new code (however I've gotta say I prefer this version's).
- Fourth, I'm not using NodeJS so I haven't debugged it for that, however I suspect it won't be working for that.

cyruscook committed:
Added test.html

cyruscook committed:
NO JQUERY

cyruscook committed:
Removed jquery dependencies

cyruscook committed:
Removed some JQuery dependencies such as extend

cyruscook committed:
Updated merge function to a better one

cyruscook committed:
Removed Jquery dependency, replaced $window.height() with window.innerHeight

cyruscook committed:
Fixed document.querySelector not giving an array

Replaced it with document.querySelectorAll

cyruscook committed:
Fixed setting heights

cyruscook committed:
Removed more JQuery dependencies when setting heights

cyruscook committed:
Added replacement for $(elem).offset()

cyruscook committed:
Replaced $(elem).on() with elem.addEventListener

cyruscook committed:
Replaced scrollTop with pageYOffset

cyruscook committed:
Fixed heights not setting

cyruscook committed:
Miscellanous syntax improvements

cyruscook committed:
Fixed margins and improved test page

lukehaas commented:
This is an amazing contribution, thank you.
I will review your PR and try it out on the example pages. If it's all good, I'll certainly want to merge this in.

cyruscook committed:
Changed test.html to tabs

cyruscook committed:
Replaced $(elem).attr with elem.getAttribute

cyruscook committed:
Next & Previous verified working

cyruscook committed:
Replaced many functions with arrow syntax

lukehaas commented:
One thing to note, as you're using modern JavaScript, we'll need to introduce Babel to transpile this for older browser support.

cyruscook committed:
Fixed issue with instant scrolling

cyruscook committed:
Oops, fixed incorrect paramers in window.scrollTo

cyruscook commented:
> to

Ah, ok, hadn't been thinking about that. Would that involve a development file and a production file? I guess that would also mean that we could compress the production file which would be beneficial.

cyruscook commented:
I believe line 917 (`if(panel.originalEvent)`) is checking if panel is an event? I'm not too sure why that is however, as if I'm not mistaken panel is meant to be either the index or name of a panel. Do you know what that's meant to be for?

cyruscook committed:
Added some comment and formatting to animateScroll()

cyruscook committed:
AnimateScroll still not working properly

cyruscook committed:
Fixed incorrectly retrieving and setting scroll distance in SizePanels

cyruscook committed:
Removed no favicon console error

lukehaas commented:
Regarding ` (if(panel.originalEvent))` I think the intention here was to enable something like this:
`$('a').click($.scrollify.move)`

But I don' think I ever documented this so it's probably not used like that by anyone. 
Happy for you to remove this.

cyruscook committed:
Added settings.logging

cyruscook committed:
Maybe allowed indexes in move()?

cyruscook committed:
Added protection against out of range panels in move()

cyruscook committed:
Merge branch 'master' of https://github.com/cyruscook/Scrollify

cyruscook committed:
Removed $window

cyruscook committed:
Removed wheelhandler JQuery dependencies

cyruscook committed:
Fully removed JQuery (most likely)

cyruscook committed:
Removed some inappropriate comments

cyruscook committed:
Moved scrollify.js -> dev.scrollify.js

cyruscook committed:
Added compressed scrollify.js

cyruscook committed:
Updated ReadMe to reflect changes

cyruscook committed:
Fixed scrollify.destroy

cyruscook committed:
Added scrollify.getOptions()

cyruscook committed:
Used beautifier.io. Hopefully formatting should now be consistent. I'll stick to this format in the future.

cyruscook committed:
Recompiled

cyruscook commented:
Wow. Incredible to see something so small from something so big.
![image](https://user-images.githubusercontent.com/11858105/57158355-1e972c80-6ddb-11e9-9c99-7c0263a35293.png)

I've compressed the file and run it through babel. I've fixed a lot of things, but I've added a `Known Issues` section to the readme. I'm currently in the process of debugging them.

cyruscook commented:
I'm having some trouble currently, could you help me?

When offset is called on page load, it returns incorrect values, which is why the page snaps. If you check the array heights, its been populated with these incorrect heights. For some reason though, it seems that running the function again makes it work - I'm guessing if this is a problem with the page not fully loading yet?

cyruscook committed:
Fixed some incorrect $ usage

lukehaas commented:
@cyruscook I'll take a look into that

lukehaas committed:
Remove duplicate extend

lukehaas committed:
Create helper methods for scroll top and now

cyruscook committed:
A few tweaks to tidy up (#1)

A few tweaks to tidy up

lukehaas committed:
add a cross-browser offset function

cyruscook committed:
add a cross-browser offset function (#2)

add a cross-browser offset function

geekymemo reviewed (commented):
hi,
.height() is jquery method for js must be .offsetHeight
if (Math.floor(elements[index].offsetHeight / portHeight) > interstitialIndex) {
